### PR TITLE
fix: read `isLegacyEnv` correctly

### PR DESCRIPTION
**Pull request by threepointone**
This fixes the signature for `isLegacyEnv()` since it doesn't use args, and we fix reading legacy_env correctly when creating a draft worker when creating a secret.

### CODE REVIEW

File: packages/wrangler/src/index.tsx
Comment (threepointone) [inline packages/wrangler/src/index.tsx:112]: @JacobMGEvans Do you remember why this change was made in https://github.com/cloudflare/wrangler2/pull/680/files#diff-5e47c8c5ec00bfdc010f06653422c5db6be55b5c61226212cf593b8c57098b6cR103 ?

**Review by caass — APPROVED** [review]
approving pending an answer on why the change u mentioned was made. makes sense otherwise

**Comment by threepointone** [comment]
yeah I'll let @JacobMGEvans have a look before I merge, thank you!

File: packages/wrangler/src/index.tsx
Comment (petebacondarwin) [reply packages/wrangler/src/index.tsx:112]: This is because the `config.legacy_env` should have already taken into account if the `legacy-env` arg was being used.

See https://github.com/cloudflare/wrangler2/blob/b24aeb5722370c2e04bce97a84a1fa1e55725d79/packages/wrangler/src/config/validation.ts#L94-L98

@threepointone - was this actually causing an error somewhere? Could you provide a test that shows that the previous code was not working?

File: packages/wrangler/src/index.tsx
Comment (petebacondarwin) [inline packages/wrangler/src/index.tsx:1468]: The change on this line looks good to me.

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Can you double check that this is actually broken?

File: packages/wrangler/src/index.tsx
Comment (threepointone) [reply packages/wrangler/src/index.tsx:112]: Oh I see. Ok I'll add it as a comment then, I was confused why it was only reading `config`. We probably don't need to pass `args` to this function either then.

**Comment by JacobMGEvans** [comment]
This function is checking for validation of configuration
```ts
const isLegacyEnv =
    (args as { "legacy-env": boolean | undefined })["legacy-env"] ??
    rawConfig.legacy_env ??
    true;
    ```
    
   This function seems to be checking against only the config which I think I missed when moving args to be generically passed to the validation check for the `normalizeAndValidateConfig` I would question if instead of removing the args from the param/argument of this helper if it shouldn't look like the validator function 
   ```ts
   function isLegacyEnv(args: unknown, config: Config): boolean {
  return config.legacy_env;
}
```

**Comment by threepointone** [comment]
It doesn't check args though, which is confusing. It's just a local helper, let's have it take only what it needs.

**Comment by JacobMGEvans** [comment]
> It doesn't check args though, which is confusing. It's just a local helper, let's have it take only what it needs.

What I am saying is it should take both and use them, since the local helper is checking which mode it is in. 
It is unclear to me why we would check `args` and `config` in the validator but not the helper checking what mode it is in locally which args is in scope of. 
```ts
function isLegacyEnv(args: unknown, config: Config): boolean {
  return  (args as { "legacy-env": boolean | undefined })["legacy-env"] ??
    config.legacy_env ??
    true;
}

File: packages/wrangler/src/index.tsx
Comment (JacobMGEvans) [inline packages/wrangler/src/index.tsx:112]:
```suggestion
  // args["legacy-env"] in https://github.com/cloudflare/wrangler2/blob/b24aeb5722370c2e04bce97a84a1fa1e55725d79/packages/wrangler/src/config/validation.ts#L94-L98
```

**Review by JacobMGEvans — APPROVED** [review]

**Comment by threepointone** [comment]
Landing this since it's now a smaller fix, and Pete's on leave and I don't want to bother him more haha.

**Comment by petebacondarwin** [comment]
The original idea was to drop the `isLegacyEnv()` function altogether but we were going to do it in a follow up PR.

**Pull request merged**
